### DISCUSSION TITLE
Renovate and esbuild

### DISCUSSION THREAD

**Discussion by Secreto31126** [renovatebot/renovate#27571] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Hi, I'm struggling to get renovate working properly with esbuild's versioning. They use the minor number to mark breaking updates, so I tried to configure renovate.json with the following rule:

```json
"packageRules": [
        {
              "packageNames": ["esbuild"],
              "versioning": "regex:^(<major>\\d+)\\.(<major>\\d+)\\.(<patch>\\d+)$"
        }
]
```

However, it seems like renovate is now ignoring all updates, and esbuild isn't even listed as a dependecy in the dashboard issue.

**Comment by rarkins** [comment]
What are you ultimately trying to achieve? Ie why change anything at all?

**Reply by Secreto31126** [reply]
I want Renovate to treat minor updates as major, so it generates a separate PR from all the other minor updates. I once accidentally merged a breaking update because I didn't notice esbuild was among the minor updated dependecies.

**Reply by rarkins** [reply]
Renovate by default separates updates from each other

**Reply by Secreto31126** [reply]
Oh, my bad, I forgot to mention I'm using "group:allNonMajor", I thought it was default behaviour. I copy pasted the config file months ago :)

My question still remains though, how can I make renovate treat esbuild middle updates as major updates?

**Reply by rarkins** [reply]
The source for that is here: https://docs.renovatebot.com/presets-group/#groupallnonmajor

Instead of using it, it would be easier to modify it slightly like this:

```
{
  "packageRules": [
    {
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch",
      "matchPackagePatterns": [
        "*"
      ],
      "excludePackageNames": ["esbuild"],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ]
    }
  ]
}
```

**Reply by Secreto31126** [reply]
Ok, I'm gonna guess this will separate all of esbuild updates, including patchs, which I don't mind, but isn't exactly what I was looking for.

**Comment by Secreto31126** [comment, marked as answer]
Found what I was looking for while rereading the docs (thanks @rarkins 😄). It isn't mentioned, but seems like the regex doesn't like having multiple groups of the same type (kinda obvious in restropective). So I changed my regex to:

`regex:^(?<compatibility>\\d+)\\.(?<major>\\d+)\\.(?<patch>\\d+)?$`

And it's working exactly as expected (separate PR for 0.19.11 -> 0.20.1).

The day esbuild releases 1.0.0 I will see what happens, but that's a future me problem :)
